### PR TITLE
Module Smart Proxy Logs

### PR DESCRIPTION
**Pull request by lzap**
Smart Proxy received a log buffer (core feature) and Logs API (core module)
which allows to download latest N log entries via REST API. We present this in
the new set of Smart Proxy status screens.

This patch aims to add configuration of the Logs API module. The module itself
has zero configuration and the only purpose is to provide simple REST API to
download the entries in JSON format.

Currently both buffer and the API are in core, but there is a pending PR to
refactor it into a module: https://github.com/theforeman/smart-proxy/pull/367

I will follow up with a Puppet changes to allow configuration of the capacity:

https://github.com/theforeman/smart-proxy/blob/develop/config/settings.yml.example#L60-L63

I was not able to run the spec tests locally, all failed with some "override"
error.

### CODE REVIEW

File: manifests/plugin/logs.pp [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# = Foreman Proxy Abrt plugin
```
Comment (ekohl) [inline manifests/plugin/logs.pp]: Please update this :)

File: manifests/plugin/logs.pp [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  $enabled                 = $::foreman_proxy::plugin::logs::params::enabled,
+  $listen_on               = $::foreman_proxy::plugin::logs::params::listen_on,
+) inherits foreman_proxy::plugin::logs::params {
+
+  validate_bool($enabled)
+  validate_listen_on($listen_on)
+
+  foreman_proxy::plugin { 'logs':
+  } ->
+  foreman_proxy::settings_file { 'logs':
```
Comment (ekohl) [inline manifests/plugin/logs.pp]: I think we should refactor the plugin definition to include a config file. I think you risk not restarting the service this way.

File: manifests/plugin/logs.pp [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    template_path => 'foreman_proxy/plugin/logs.yml.erb',
+    group         => $group,
```
Comment (ekohl) [inline manifests/plugin/logs.pp]: This is undefined in this scope.

**Comment by domcleal** [comment]
This PR seems to be configuring a *plugin*, is that correct?  That's different to a config file shipped by default.

> I will follow up with a Puppet changes to allow configuration of the capacity:
> 
> https://github.com/theforeman/smart-proxy/blob/develop/config/settings.yml.example#L60-L63

Please see PR #214 for this.

File: manifests/plugin/logs.pp (hunk 2 above)
Comment (ekohl) [inline manifests/plugin/logs.pp]: Never mind that, settings_file does that.

File: manifests/plugin/logs.pp [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+#
+class foreman_proxy::plugin::logs (
+  $enabled                 = $::foreman_proxy::plugin::logs::params::enabled,
+  $listen_on               = $::foreman_proxy::plugin::logs::params::listen_on,
+) inherits foreman_proxy::plugin::logs::params {
+
+  validate_bool($enabled)
+  validate_listen_on($listen_on)
+
+  foreman_proxy::plugin { 'logs':
```
Comment (ekohl) [inline manifests/plugin/logs.pp]: Since this is a module in core, I don't think this will be packaged separately and thus not needed.

**Comment by lzap** [comment]
Yeah, my intention is to cover the patch that introduces new *module*: https://github.com/theforeman/smart-proxy/pull/367. If my understanding is correct: *module* = *plugin* shipped with core.

Anyway, this is my attempt to allow users to enable/disable Logs REST API.

**Comment by domcleal** [comment]
> If my understanding is correct: module = plugin shipped with core.

No, plugins are additional modules installed through packages.  Anything provided in core should always be configured in manifests/config.pp.

**Comment by lzap** [comment]
Ok I think I can close this patch then. I can make an attempt unless you plan to cover that. I don't know what I am doing in Puppet really.

**Comment by domcleal** [comment]
Ok, I'll probably have a go after any changes are made in the smart proxy itself, but please feel free to file a ticket under the installer Redmine project to remind us for the release it's included in.

**Comment by lzap** [comment]
Let me try this instead: https://github.com/theforeman/puppet-foreman_proxy/pull/217

**Pull request closed without merging**